### PR TITLE
new proposed locators

### PR DESCRIPTION
**Pull request by Enriquesuarezgtte**

### CODE REVIEW

File: src/page/address-step.page.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { $, ElementFinder, browser, ExpectedConditions } from 'protractor';
+import { ElementFinder, browser, ExpectedConditions, by, element } from 'protractor';
 
 export class AddressStepPage {
   private cartAddressNavigation: ElementFinder;
 
   constructor() {
-    this.cartAddressNavigation = $('#center_column > form > p > button > span');
+    this.cartAddressNavigation = element(by.name('processAddress'));
```
Comment (leonleo997) [inline src/page/address-step.page.ts]: try to use `element `or `$()` to get the elements but avoid to use both

File: src/page/menu-content.page.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ export class MenuContentPage {
   private tShirtMenu: ElementFinder;
 
   constructor() {
-    this.tShirtMenu = $('#block_top_menu > ul > li:nth-child(3) > a');
+    this.tShirtMenu = $('#block_top_menu .sf-menu > li:nth-last-child(1)');
```
Comment (leonleo997) [inline src/page/menu-content.page.ts]: This will fail if the order is changed, try to use a better locator

File: src/page/payment-step.page.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
-import { $, ElementFinder, browser, ExpectedConditions } from 'protractor';
+import { ElementFinder, browser, ExpectedConditions, by, element } from 'protractor';
 
 export class PaymentStepPage {
   private payByCheckOption: ElementFinder;
 
   constructor() {
-    this.payByCheckOption = $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a');
+    this.payByCheckOption = element(by.className('cheque'));
```
Comment (leonleo997) [inline src/page/payment-step.page.ts]: same here

File: src/page/product-list.page.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ export class ProductListPage {
   private addToCartItem: ElementFinder;
 
   constructor() {
-    this.addToCartItem = $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default');
+    this.addToCartItem = $('.button-container a.button.ajax_add_to_cart_button.btn.btn-default');
```
Comment (leonleo997) [inline src/page/product-list.page.ts]: try to find a better locator

File: src/page/shipping-step.page.ts [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+import { $, ElementFinder, browser, ExpectedConditions, element, by } from 'protractor';
 
 export class ShippingStepPage {
   private termsOfServiceCheck: ElementFinder;
   private shippingNavigation: ElementFinder;
 
   constructor() {
-    this.termsOfServiceCheck = $('#cgv');
-    this.shippingNavigation = $('#form > p > button > span');
+    this.termsOfServiceCheck = element(by.id('cgv'));
```
Comment (leonleo997) [inline src/page/shipping-step.page.ts]: This is a good locator but same that above, try to use element or $() but avoid to merge them

File: src/page/sign-in-step.page.ts [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
 export class SignInCardStepPage {
   private emailField: ElementFinder;
   private passwordField: ElementFinder;
   private signInOption: ElementFinder;
 
   constructor() {
-    this.emailField = $('#email');
-    this.passwordField = $('#passwd');
-    this.signInOption = $('#SubmitLogin > span');
+    this.emailField = element(by.id('email'));
```
Comment (leonleo997) [inline src/page/sign-in-step.page.ts]: these are good locators but the same that above, try to use element or $() but avoid to merge them

File: src/page/shipping-step.page.ts [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
+    this.shippingNavigation = $('.cart_navigation .standard-checkout');
```
Comment (leonleo997) [inline src/page/shipping-step.page.ts]: there is no a better locator?

File: src/page/order-result.page.ts [hunk 8]
```diff
@@ -4,7 +4,7 @@ export class OrderResultPage {
   private orderResultAlert: ElementFinder;
 
   constructor() {
```
Comment (leonleo997) [inline src/page/order-result.page.ts]: is this page a real page object? or should be part of another one?

File: src/page/menu-content.page.ts (hunk 2 above)
Comment (Enriquesuarezgtte) [reply src/page/menu-content.page.ts]: You're right

File: src/page/address-step.page.ts (hunk 1 above)
Comment (Enriquesuarezgtte) [reply src/page/address-step.page.ts]: Ok. I just found $( )  is a shortcut to element( by.css(  ) )

File: src/page/shipping-step.page.ts (hunk 7 above)
Comment (Enriquesuarezgtte) [reply src/page/shipping-step.page.ts]: I found one

File: src/page/order-result.page.ts (hunk 8 above)
Comment (Enriquesuarezgtte) [reply src/page/order-result.page.ts]: You made me realize this page object is part from Payment step. However, I thought it was not by reloaded page after confirm order.

**Comment by Enriquesuarezgtte** [comment]
> 
> 
> take a look of this section of [selecting elements](https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements)

Thanks for this advice, was incredibly useful

**Pull request merged**